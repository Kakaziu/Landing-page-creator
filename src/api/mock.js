export const pageData = [
  {
    id: 1,
    attributes: {
      title: 'PÃ¡gina 1',
      slug: 'page-1',
      footer_text: [
        {
          type: 'paragraph',
          children: [
            {
              type: 'text',
              text: 'footer',
            },
          ],
        },
      ],
      sections: [
        {
          id: 1,
          __component: 'section.section-two-columns',
          title: 'Section two columns',
          description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu porttitor augue, ut auctor sem. Cras rhoncus pellentesque ante non pharetra. Maecenas pretium consectetur sapien pellentesque rutrum. Aliquam vitae tempor purus, at rhoncus orci. Suspendisse a sem non dolor sagittis mattis.',
          image: {
            data: {
              id: 1,
              attributes: {
                alternativeText: null,
                url: 'https://res.cloudinary.com/drplqaxi0/image/upload/v1704251425/pexels_angel_rondon_19483671_99d3895603.jpg',
              },
            },
          },
          button: {
            id: 1,
            text_button: 'Clique aqui',
            url: '#',
            size: 'big',
          },
          metadata: {
            id: 1,
            name: 'section-two-columns-1',
            section_id: 'hero',
            background: true,
          },
        },
        {
          id: 2,
          __component: 'section.section-two-columns',
          title: 'Sobre',
          description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu porttitor augue, ut auctor sem. Cras rhoncus pellentesque ante non pharetra. Maecenas pretium consectetur sapien pellentesque rutrum. Aliquam vitae tempor purus, at rhoncus orci. Suspendisse a sem non dolor sagittis mattis.',
          image: {
            data: {
              id: 1,
              attributes: {
                alternativeText: null,
                url: 'https://res.cloudinary.com/drplqaxi0/image/upload/v1704251425/pexels_angel_rondon_19483671_99d3895603.jpg',
              },
            },
          },
          button: null,
          metadata: {
            id: 2,
            name: 'section-two-columns-2',
            section_id: 'sobre',
            background: false,
          },
        },
        {
          id: 1,
          __component: 'section.section-grid',
          title: 'grid',
          description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu porttitor augue, ut auctor sem. Cras rhoncus pellentesque ante non pharetra. Maecenas pretium consectetur sapien pellentesque rutrum. Aliquam vitae tempor purus, at rhoncus orci. Suspendisse a sem non dolor sagittis mattis.',
          button: {
            id: 3,
            text_button: 'aaaaa clica',
            url: '#',
            size: 'big',
          },
          metadata: {
            id: 3,
            name: 'grid-1',
            section_id: 'depoimentos',
            background: true,
          },
          text_grid: [
            {
              id: 1,
              title: 'grid 1',
              description:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.',
            },
            {
              id: 2,
              title: 'grid 2',
              description:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.',
            },
            {
              id: 3,
              title: 'grid 3',
              description:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.',
            },
          ],
          image_grid: [],
        },
        {
          id: 1,
          __component: 'section.section-content',
          title: 'contato',
          content: [
            {
              type: 'paragraph',
              children: [
                {
                  type: 'text',
                  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ',
                },
                {
                  type: 'text',
                  text: 'ipsum dolor sit amet',
                  bold: true,
                },
                {
                  type: 'text',
                  text: ', consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.',
                },
              ],
            },
          ],
          button: {
            id: 4,
            text_button: 'botao',
            url: '#',
            size: 'big',
          },
          metadata: {
            id: 4,
            name: 'section-content',
            section_id: 'contato',
            background: true,
          },
        },
        {
          id: 1,
          __component: 'section.section-box',
          title: 'Section Box',
          button: null,
          paragraph: [
            {
              id: 1,
              text: "here are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.",
            },
            {
              id: 2,
              text: "here are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.",
            },
            {
              id: 3,
              text: "here are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.",
            },
          ],
          section_metadata: {
            id: 5,
            name: 'section_box',
            section_id: 'vvvvvv',
            background: false,
          },
        },
      ],
      menu: {
        id: 1,
        logo_title: 'logo',
        logo_link: '#',
        image_logo: {
          data: {
            id: 1,
            attributes: {
              url: 'a.svg',
            },
          },
        },
        menu_links: [
          {
            id: 1,
            text: 'Home',
            url: '#home',
            open_in_new_tab: false,
          },
          {
            id: 2,
            text: 'Sobre',
            url: '#sobre',
            open_in_new_tab: false,
          },
          {
            id: 3,
            text: 'Depoimentos',
            url: '#depoimentos',
            open_in_new_tab: false,
          },
          {
            id: 4,
            text: 'Contato',
            url: '#contato',
            open_in_new_tab: false,
          },
        ],
      },
    },
  },
];

export const dataMenu = {
  id: 1,
  logo_title: 'logo',
  logo_link: '#',
  image_logo: {
    data: {
      id: 1,
      attributes: {
        url: 'a.svg',
      },
    },
  },
  menu_links: [
    {
      id: 1,
      text: 'Home',
      url: '#home',
      open_in_new_tab: false,
    },
    {
      id: 2,
      text: 'Sobre',
      url: '#sobre',
      open_in_new_tab: false,
    },
    {
      id: 3,
      text: 'Depoimentos',
      url: '#depoimentos',
      open_in_new_tab: false,
    },
    {
      id: 4,
      text: 'Contato',
      url: '#contato',
      open_in_new_tab: false,
    },
  ],
};

export const menuLinks = [
  {
    id: 1,
    text: 'Home',
    url: '#home',
    open_in_new_tab: false,
  },
  {
    id: 2,
    text: 'Sobre',
    url: '#sobre',
    open_in_new_tab: false,
  },
  {
    id: 3,
    text: 'Depoimentos',
    url: '#depoimentos',
    open_in_new_tab: false,
  },
  {
    id: 4,
    text: 'Contato',
    url: '#contato',
    open_in_new_tab: false,
  },
];

export const sectionTwoColumnsMock = {
  id: 1,
  __component: 'section.section-two-columns',
  title: 'Section two columns',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu porttitor augue, ut auctor sem. Cras rhoncus pellentesque ante non pharetra. Maecenas pretium consectetur sapien pellentesque rutrum. Aliquam vitae tempor purus, at rhoncus orci. Suspendisse a sem non dolor sagittis mattis.',
  image: {
    data: {
      id: 1,
      attributes: {
        alternativeText: 'image',
        url: 'b.svg',
      },
    },
  },
  button: {
    id: 1,
    text_button: 'Clique aqui',
    url: '#',
    size: 'big',
  },
  metadata: {
    id: 1,
    name: 'section-two-columns-1',
    section_id: 'hero',
    background: true,
  },
};

export const sectionBoxMock = {
  id: 1,
  __component: 'section.section-box',
  title: 'Section Box',
  button: null,
  paragraph: [
    {
      id: 1,
      text: 'paragraph',
    },
    {
      id: 2,
      text: "here are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.",
    },
    {
      id: 3,
      text: "here are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.",
    },
  ],
  section_metadata: {
    id: 5,
    name: 'section_box',
    section_id: 'vvvvvv',
    background: false,
  },
};
