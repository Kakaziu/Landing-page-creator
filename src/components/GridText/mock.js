export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  gridText: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
  ],
  button: {
    textButton: 'Clique aqui',
    urlButton: '#',
    size: 'small',
  },
};
