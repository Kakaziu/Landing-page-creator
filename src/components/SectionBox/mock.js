export default {
  title: 'Section Box',
  paragraphs: [
    {
      text: 'There are 1',
    },
    {
      text: 'There are 2',
    },
    {
      text: 'There are 3',
    },
  ],
  button: {
    textButton: 'Clique aqui',
    urlButton: '#',
    size: 'small',
  },
};
