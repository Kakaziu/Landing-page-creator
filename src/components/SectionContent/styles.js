import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

export const TitleContainer = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
      text-align: center;
    }
  `}
`;

export const HtmlContainer = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.small};
    text-align: center;

    p {
      margin-top: ${theme.spacings.medium}
    }
  `}
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
  ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
      color: ${background ? theme.colors.white : theme.colors.primaryColor}
    }
  `}
`;
