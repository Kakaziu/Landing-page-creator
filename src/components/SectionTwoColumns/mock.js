export default {
  title: 'Grid two columns',
  text: 'Alo hamigos',
  srcImg: './assets/images/image1.svg',
  button: {
    textButton: 'Clique aqui',
    urlButton: '#',
    size: 'big',
  },
};
