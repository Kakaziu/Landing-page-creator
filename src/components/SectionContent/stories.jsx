import SectionContent from '.';
import mock from './mock';

export default {
  title: 'SectionContent',
  component: SectionContent,
  args: mock,
};

export const Template = (args) => (
  <div>
    <SectionContent {...args} />
  </div>
);
