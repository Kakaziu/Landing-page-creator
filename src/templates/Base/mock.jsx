import links from '../../components/NavLink/mock';
import GridText from '../../components/GridText';
import gridMock from '../../components/GridText/mock';

export default {
  children: (
    <>
      <GridText {...gridMock} background />
      <GridText {...gridMock} />
      <GridText {...gridMock} background />
      <GridText {...gridMock} />
      <GridText {...gridMock} background />
      <GridText {...gridMock} />
    </>
  ),
  links: links,
  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml: '<p>Teste de footer</p>',
};
