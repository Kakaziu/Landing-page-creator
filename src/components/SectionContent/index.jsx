import P from 'prop-types';
import SectionBackground from '../SectionBackground';
import * as Styled from './styles';
import Heading from '../Heading';
import Button from '../Button';

const SectionContent = ({
  title,
  html,
  background = false,
  sectionId = '',
  button,
}) => {
  return (
    <SectionBackground background={background}>
      <Styled.Container background={+background} id={sectionId}>
        <Styled.TitleContainer>
          <Heading uppercase>{title}</Heading>
        </Styled.TitleContainer>
        <Styled.HtmlContainer dangerouslySetInnerHTML={{ __html: html }} />
        {button && button.children && (
          <Button url={button.urlButton} size={button.size}>
            {button.children}
          </Button>
        )}
      </Styled.Container>
    </SectionBackground>
  );
};

SectionContent.propTypes = {
  title: P.string.isRequired,
  html: P.string.isRequired,
  background: P.bool,
  sectionId: P.string,
  ...Button.propTypes,
};

export default SectionContent;
