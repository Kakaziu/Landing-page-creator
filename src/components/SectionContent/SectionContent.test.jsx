import { screen } from '@testing-library/react';
import SectionContent from '.';
import mock from './mock';
import renderTheme from '../../styles/render-theme';

describe('<SectionContent/>', () => {
  it('should render SectionContent', () => {
    const { container } = renderTheme(<SectionContent {...mock} />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render SectionContent with background', () => {
    renderTheme(<SectionContent {...mock} background={true} />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });
});
