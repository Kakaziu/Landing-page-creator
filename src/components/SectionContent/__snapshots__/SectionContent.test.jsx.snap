// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionContent/> should render SectionContent 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c5 {
  color: #0A1128;
  text-transform: uppercase;
  font-size: 3.2rem;
}

.c8 {
  display: block;
}

.c9 {
  font-weight: bold;
  border: none;
  background-color: #dc143c;
  color: #FFFFFF;
  border-radius: 4px;
  padding: 8px 40px;
  font-size: 2.4rem;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

.c9:hover {
  background-color: #b81132;
}

.c3 .c4 {
  margin-bottom: 1.6rem;
  text-align: center;
}

.c6 {
  font-size: 1.6rem;
  text-align: center;
}

.c6 p {
  margin-top: 2.4rem;
}

.c2 .c4 {
  margin-bottom: 1.6rem;
  color: #0A1128;
}

.c2 .c7 {
  margin-top: 2.4rem;
  text-align: center;
}

<div>
  <div
    background="0"
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        background="0"
        class="c2"
      >
        <div
          class="c3"
        >
          <h1
            class="c4 c5"
            isdark="1"
            uppercase="1"
          >
            TÃ­tulo
          </h1>
        </div>
        <div
          class="c6"
        >
          
  
          <p>
            Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra. Vestibulum erat wisi, condimentum sed, commodo vitae, ornare sit amet, wisi. Aenean fermentum, elit eget tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. Donec non enim in turpis pulvinar facilisis. Ut felis. Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus
          </p>
          
  
          <p>
            Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra. Vestibulum erat wisi, condimentum sed, commodo vitae, ornare sit amet, wisi. Aenean fermentum, elit eget tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. Donec non enim in turpis pulvinar facilisis. Ut felis. Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus
          </p>
          
  
        </div>
        <a
          class="c7 c8"
          href="#"
        >
          <button
            class="c9"
          >
            Clique aqui
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
